third-party/CMakeLists.txt

```diff
@@ -0,0 +1,247 @@
+# Copyright (c) 2014-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed in accordance with the terms specified in
+# the LICENSE file found in the root directory of this source tree.
+
+cmake_minimum_required(VERSION 3.13.3)
+
+function(osqueryThirdpartyMain)
+  initializePythonPathFolder()
+
+  add_subdirectory("augeas")
+  add_subdirectory("aws-sdk-cpp")
+  add_subdirectory("berkeley-db")
+  add_subdirectory("boost")
+  add_subdirectory("bzip2")
+  add_subdirectory("gflags")
+  add_subdirectory("glibc")
+  add_subdirectory("glog")
+  add_subdirectory("googletest")
+  add_subdirectory("jinja2")
+  add_subdirectory("libarchive")
+  add_subdirectory("libaudit")
+  add_subdirectory("libcryptsetup")
+  add_subdirectory("libdevmapper")
+  add_subdirectory("libdpkg")
+  add_subdirectory("libelfin")
+  add_subdirectory("libgcrypt")
+  add_subdirectory("libgpg-error")
+  add_subdirectory("libiptables")
+  add_subdirectory("libmagic")
+  add_subdirectory("librdkafka")
+  add_subdirectory("librpm")
+  add_subdirectory("libudev")
+  add_subdirectory("libxml2")
+  add_subdirectory("linenoise-ng")
+  add_subdirectory("lldpd")
+  add_subdirectory("lzma")
+  add_subdirectory("markupsafe")
+  add_subdirectory("openssl")
+  add_subdirectory("popt")
+  add_subdirectory("rapidjson")
+  add_subdirectory("rocksdb")
+  add_subdirectory("sleuthkit")
+  add_subdirectory("smartmontools")
+  add_subdirectory("sqlite")
+  add_subdirectory("ssdeep-cpp")
+  add_subdirectory("thrift")
+  add_subdirectory("util-linux")
+  add_subdirectory("yara")
+  add_subdirectory("zlib")
+  add_subdirectory("zstd")
+endfunction()
+
+# Generates a target named identifier_downloader that will acquire the remote file while also verifying
+# its hash
+function(downloadRemoteFile identifier base_url file_name hash)
```

Comment:
I think we can remove a lot of the helper code here if we replace the dependency management in the future.